lib/mime.ex

```diff
@@ -164,6 +164,13 @@ defmodule MIME do
     unquote(Macro.escape(custom_types))
   end
 
+  @doc """
+  Returns a list of all known types, including custom types compiled into the MIME module.
+  """
+  def known_types do
+    Map.keys(unquote(Macro.escape(all_types)))
```

Comment:
Wouldn't it be better to compile them with their extensions, so all information becomes available?

Reply:
I made it return just the types because that's what I would expect a `known_types` function to do but it's also better to have more information. I changed it to return the mapping of types to extensions. :)